LibVLCSharp/Platforms/UWP/VideoView.cs

```diff
@@ -0,0 +1,233 @@
+                Format = SharpDX.DXGI.Format.B8G8R8A8_UNorm,
+                Stereo = false,
+                SampleDescription =
+                {
+                    Count = 1,
+                    Quality = 0
+                },
+                Usage = Usage.RenderTargetOutput,
+                BufferCount = 2,
+                SwapEffect = SwapEffect.FlipSequential,
+                Flags = SwapChainFlags.None,
+                AlphaMode = AlphaMode.Unspecified
+            };
+
+            _swapChain = new SharpDX.DXGI.SwapChain1(dxgiFactory, _d3d11Device, ref swapChainDescription);
+
+            _device.MaximumFrameLatency = 1;
+
+            //TODO: perform the next 2 calls on the UI thread
+            var panelNative = ComObject.As<ISwapChainPanelNative>(this._panel);
+            panelNative.SwapChain = _swapChain;
+
+            // This is necessary so we can call Trim() on suspend
+            this._device3 = _device.QueryInterface<SharpDX.DXGI.Device3>();
+            if (_device3 == null)
+                throw new Exception();
+            this._swapchain2 = _swapChain.QueryInterface<SharpDX.DXGI.SwapChain2>();
+            if (_swapchain2 == null)
+                throw new Exception();
+
+            UpdateScale();
+            UpdateSize();
+            this._loaded = true;
+        }
+
+        void OnUnload()
+        {
+            this._swapchain2?.Dispose();
+            this._swapchain2 = null;
+
+            this._device3?.Dispose();
+            this._device3 = null;
+
+            this._swapChain?.Dispose();
+            this._swapChain = null;
+
+            this._device?.Dispose();
+            this._device = null;
+
+            this._d3d11Device?.Dispose();
+            this._d3d11Device = null;
+        }
+
+        readonly Guid SWAPCHAIN_WIDTH = new Guid(0xf1b59347, 0x1643, 0x411a, 0xad, 0x6b, 0xc7, 0x80, 0x17, 0x7a, 0x6, 0xb6);
+        readonly Guid SWAPCHAIN_HEIGHT = new Guid(0x6ea976a0, 0x9d60, 0x4bb7, 0xa5, 0xa9, 0x7d, 0xd1, 0x18, 0x7f, 0xc9, 0xbd);
+
+        void UpdateSize()
+        {
+            IntPtr width = Marshal.AllocHGlobal(sizeof(int));
+            IntPtr height = Marshal.AllocHGlobal(sizeof(int));
+            var w = (int)(this._panel.ActualWidth * this._panel.CompositionScaleX);
+            var h = (int)(this._panel.ActualHeight * this._panel.CompositionScaleY);
+
+            Marshal.WriteInt32(width, w);
+            Marshal.WriteInt32(height, h);
+
+            _swapChain.SetPrivateData(SWAPCHAIN_WIDTH, sizeof(int), width);
+            _swapChain.SetPrivateData(SWAPCHAIN_HEIGHT, sizeof(int), height);
+
+            Marshal.FreeHGlobal(width);
+            Marshal.FreeHGlobal(height);
+        }
+
+        void UpdateScale()
+        {
+            _swapchain2.MatrixTransform = new RawMatrix3x2 { M11 = 1.0f / this._panel.CompositionScaleX, M22 = 1.0f / this._panel.CompositionScaleY };
+        }
+
+        void Trim()
```

Comment:
Oh, I forgot to call that. We need to do that when the app is suspending.

Reply:
Yeah. Don't know if we can do that from library code without custom user callback

Reply:
We can. Done.